### ISSUE TITLE
Crash when executing machine code, but not when interpreting

### ISSUE THREAD

**Issue by edubart** [vnmakarov/mir#185]
The following  example crashes with `-eg` but not with `-ei` when running with C2M.

```c
#include <stdint.h>
#include <stddef.h>
#include <stdio.h>

typedef struct string {
  uint8_t* data;
  uintptr_t size;
} string;

typedef struct niltype {} niltype;
#define NIL (niltype){}

typedef struct Foo {
  void* impl;
  niltype allocator;
} Foo;

typedef struct Tuple {
  int r1;
  Foo r2;
} Tuple;

static Tuple foo(string pattern, niltype init) {
  Tuple mulret = {0};
  printf("%s\n", pattern.data);
  return mulret;
}
int main(int argc, char** argv) {
  // Tuple x = {0};
  Tuple _asgnret_1 = foo(((string){(uint8_t*)"test", 4}), NIL);
  return 0;
}
```

```
➜  c2m test.c -ei
draft.c:10:9: warning -- empty struct/union
draft.c:30:59: warning -- empty initializer list
test
➜  c2m test.c -eg
draft.c:10:9: warning -- empty struct/union
draft.c:30:59: warning -- empty initializer list
Segmentation fault (core dumped)
```

Uncommenting the line `// Tuple x = {0};`, the crash goes away, but behaves different (odd) than interpreting:
```c
➜  c2m test.c -ei
draft.c:10:9: warning -- empty struct/union
draft.c:30:59: warning -- empty initializer list
test
➜  c2m test.c -eg
draft.c:10:9: warning -- empty struct/union
draft.c:30:59: warning -- empty initializer list
(null)
```

Of course this must be something going on with empty structs, thus #183 may be related.
My environment is latest commit, GCC and Linux x86_64.

**Comment by edubart** [comment]
I noticed a fix for issue #183 was made just at the time I submitted this issue, and the fix does seem to fix the crash here, so I think you can close this if there is nothing new here.

**Comment by vnmakarov** [comment]
Yes, the code works now with -eg after fixing issue 183.  So I am closing the issue.

It is interesting that you found such a buggy C code and csmith did not find something analogous.  Recently I used 10,000 tests generated by csmith (it took a day) and compared it with results of GCC and Clang and there are no differences (code of generated by c2m/GCC/Clang fails or produces the same results).  Probably csmith does not generate empty structures.
